File: plain/asker/folder.ul.js


// try to ask for folder ul, by zmq
// 2016 0409
//


var u = require("underscore");
var zmq = require('zmq');

var localhost5555 = require("./lh5.js");

var p = console.log;

function  folder_ul(who, fopath, callback){
    var info = JSON.stringify( {
        "who": who,
        "ask-for": 'folder.ul',
        "path": fopath,
        "timeout": 3000
    });

    localhost5555.ask(info, callback);
}



process.on('SIGINT', function() {
  requester.close();
});


function check_0411(name, what){
    name = name || 'unknown';
    what = what || 'some/folder/ul';

    folder_ul(name, what, function(err, rep){
        p(err, rep);
    });
    setTimeout(process.exit, 5000);
}

if(require.main === module){
    check_0411('tmp', 'tmp/public');
}

File: python/zmqueue/folder_ul.py

import path_setting #d

#import s3.folder.getter as getter
from s3.folder import  getter


def cached_reply(info):
    if 'path' not in info:
        return {'got no path for folder ul, in function reply': True}

    #get folder ul
    asked  = info['ask-for']
    folder = getter.folder(info['path'])
    cache  = folder.get_cache()
    if 'renders' in cache and 'ul' in cache['renders']:
        info[asked] = cache['renders']['ul']
    else:
        info[asked] = -1

    return info


if __name__ == "__main__":
    info = {
            "who": 'tmp',
            "ask-for": 'folder.ul',
            "path": 'tmp/public',
            "timeout": 3000
            }

    #print cached_reply(info)
